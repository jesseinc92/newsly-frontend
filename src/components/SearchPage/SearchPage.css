.SearchPage-form input,
.SearchPage-form select {
  font-size: 1.25rem;
  font-family: 'taviraj', serif;
  padding-left: 0.5rem;
  margin-top: 1rem;
  margin-bottom: 2rem;
}

/* small desktop/laptop sizing */
@media screen and (min-width: 1024px) {

  .FrontPage h2 {
    font-size: clamp(48px, 4.68vw, 64px);
  }

  .SearchPage-results {
    width: 75%;
    margin: auto;
    margin-bottom: 2rem;

    display: grid;
    grid-template-columns: repeat(2, 1fr);
    gap: 2rem;
  }

  .SearchPage-loader {
    grid-column: 1 / 3;
  }

}

/* Large desktop/laptop sizing */
@media screen and (min-width: 1440px) {
  
  .SearchPage-results {
    grid-template-columns: repeat(3, 1fr);
  }

  .SearchPage-loader {
    grid-column: 1 / 4;
  }

}