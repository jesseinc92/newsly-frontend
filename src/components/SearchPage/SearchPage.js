import React, { useEffect, useState } from "react";
import NewslyAPI from "../../api";
import usePagination from "../../hooks/usePagination";
import './SearchPage.css';
import ArticleCard from "../ArticleCard/ArticleCard";
import Button from "../Button/Button";

const SearchPage = () => {
  const [searchData, setSearchData] = useState({
    filter: 'keyword',
    term: ''
  });
  const [searchResults, setSearchResults] = useState([]);
  const [pageNum, handlePagination] = usePagination();

  useEffect(() => {
    // retrieve saved search session on component mount.
    const savedSearch = JSON.parse(sessionStorage.getItem('sessionSearch'));
    if (savedSearch) {
      setSearchResults(savedSearch);
    }
  }, []);

  const handleChange = e => {
    const { name, value } = e.target;
    setSearchData(data => ({
      ...data,
      [name]: value
    }));
  }

  const handleSubmit = async (e) => {
    e.preventDefault();
    
    try {
    // make the search call once the form is submitted.
    const results = await NewslyAPI.searchArticles(searchData.term, searchData.filter);

    // save search results in session for later component visits.
    sessionStorage.setItem('sessionSearch', JSON.stringify(results));
    setSearchResults(results);
    } catch(err) {
      alert('Search found no results.')
    }
  }

  return (
    <main className='SearchPage'>
      <form className='SearchPage-form' onSubmit={handleSubmit}>
        <select name='filter' onChange={handleChange}>
          <option value='keyword'>Keyword</option>
          <option value='section'>Section</option>
        </select>
        <input 
          type='text'
          name='term'
          placeholder='Search for articles...' 
          value={searchData.term ?? ''} 
          onChange={handleChange} 
        />
      </form>
      <section className='SearchPage-results'>
        {searchResults.length > 0 ?
            searchResults.map(result => (
              <ArticleCard key={result.id} article={result}/>
            ))
          :
            <div className='SearchPage-loader'>
              <h1>&hellip;</h1>
            </div>
        }
      </section>

      { searchResults.length > 0 ?
        <div className='button-group page-count'>
            <Button text='previous' handler={handlePagination} />
            <p>Page { pageNum }</p>
            <Button text='next' handler={handlePagination} />
        </div>
        :
        <></>
      }
    </main>
  )
}

export default SearchPage;