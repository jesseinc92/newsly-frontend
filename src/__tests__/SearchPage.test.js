import React from 'react';
import { render } from '@testing-library/react';
import SearchPage from '../components/SearchPage/SearchPage';

it('renders the component', function() {
  render(<SearchPage />);
});

it('matches the snapshot', function() {
  const { asFragment } = render(<SearchPage />);
  expect(asFragment()).toMatchSnapshot();
});