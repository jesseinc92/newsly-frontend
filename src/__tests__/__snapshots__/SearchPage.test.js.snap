// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <main
    class="SearchPage"
  >
    <form
      class="SearchPage-form"
    >
      <select
        name="filter"
      >
        <option
          value="keyword"
        >
          Keyword
        </option>
        <option
          value="section"
        >
          Section
        </option>
      </select>
      <input
        name="term"
        placeholder="Search for articles..."
        type="text"
        value=""
      />
    </form>
    <section
      class="SearchPage-results"
    >
      <div
        class="SearchPage-loader"
      >
        <h1>
          â€¦
        </h1>
      </div>
    </section>
  </main>
</DocumentFragment>
`;
